layers/ip6.go

```diff
@@ -34,6 +75,7 @@ type IPv6 struct {
 	HopLimit     uint8
 	SrcIP        net.IP
 	DstIP        net.IP
+	RoutingDstIP net.IP
```

Comment:
Instead of a new field here, I think it would be best to make this a function:

func (i *IPv6) RoutingDstIP() net.IP {
  ... walk down IPv6 options, look for routing options, etc ...
}

Since this is a logical part of the packet and not something actually encoded therein.

Also, I think that by not having this be a field, we can remove the requirement for the new LastNetworkLayer() function in the PacketBuilder.